### DISCUSSION TITLE
Sort by relationship

### DISCUSSION THREAD

**Discussion by zolkos-ap** [aminalaee/sqladmin#313] [Q&A] [answered]
Hi,
is it possible to sort a column by a relationship attribute?

My specific case:

    class Project(Base):
        __tablename__ = 'projects'

        id = Column(UUID(as_uuid=True), nullable=False, primary_key=True, default=uuid.uuid4)
        name = Column(String, nullable=False, unique=True)

        categories = relationship("Category", back_populates="project")

        def __str__(self):
            return self.name

    class Category(Base):
        __tablename__ = 'categories'

        id = Column(UUID(as_uuid=True), nullable=False, primary_key=True, default=uuid.uuid4)
        project_id = Column(UUID(as_uuid=True), ForeignKey("projects.id"), nullable=False)

        project = relationship("Project", back_populates="categories", foreign_keys="Category.project_id", lazy="selectin")

In my CategoryAdmin ModelView I have tried:
1. `column_sortable_list = [Category.project]` but get `Mapped instance expected for relationship comparison to object.` at runtime.
2.  `column_sortable_list = [Category.project.name]` but get `Neither 'InstrumentedAttribute' object nor 'Comparator' object associated with Category.project has an attribute 'name'`

I have also defined a `__str__` method for my Project class, But that didn't resolve this issue.
Is there any way I can make the `Category.project` column sortable?

**Comment by aminalaee** [comment, marked as answer]
Hey,

The first approach is not clear to me, how do you want to sort by `Category.project`? It should be with some attribute, right?

The second approach is not possible in SQLAlchemy, so you can't do:

```py
column_sortable_list = [Category.project.name]
```

It should be supported with `column_sortable_list = ["project.name"]` but that's not available yet. Feel free to add this feature otherwise I'll get to it later.
